### PR TITLE
Adjustment for Python 3.10

### LINKED ISSUE #808: pip install failure for Python 3.10
## 🐛 Bug

pip fails to install required dependencies

### To Reproduce

```shell
 % python --version
Python 3.10.2
 % pip freeze
 % pip install torchmetrics
Collecting torchmetrics
  Using cached torchmetrics-0.7.0-py3-none-any.whl (396 kB)
  Using cached torchmetrics-0.6.2-py3-none-any.whl (332 kB)
  Using cached torchmetrics-0.6.1-py3-none-any.whl (332 kB)
  Using cached torchmetrics-0.6.0-py3-none-any.whl (329 kB)
  Using cached torchmetrics-0.5.1-py3-none-any.whl (282 kB)
  Using cached torchmetrics-0.5.0-py3-none-any.whl (272 kB)
  Using cached torchmetrics-0.4.1-py3-none-any.whl (234 kB)
  Using cached torchmetrics-0.3.2-py3-none-any.whl (274 kB)
  Using cached torchmetrics-0.3.1-py3-none-any.whl (271 kB)
  Using cached torchmetrics-0.3.0-py3-none-any.whl (270 kB)
  Using cached torchmetrics-0.2.0-py3-none-any.whl (176 kB)
ERROR: Cannot install torchmetrics==0.2.0, torchmetrics==0.3.0, torchmetrics==0.3.1, torchmetrics==0.3.2, torchmetrics==0.4.1, torchmetrics==0.5.0, torchmetrics==0.5.1, torchmetrics==0.6.0, torchmetrics==0.6.1, torchmetrics==0.6.2 and torchmetrics==0.7.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    torchmetrics 0.7.0 depends on torch>=1.3.1
    torchmetrics 0.6.2 depends on torch>=1.3.1
    torchmetrics 0.6.1 depends on torch>=1.3.1
    torchmetrics 0.6.0 depends on torch>=1.3.1
    torchmetrics 0.5.1 depends on torch>=1.3.1
    torchmetrics 0.5.0 depends on torch>=1.3.1
    torchmetrics 0.4.1 depends on torch>=1.3.1
    torchmetrics 0.3.2 depends on torch>=1.3.1
    torchmetrics 0.3.1 depends on torch>=1.3.1
    torchmetrics 0.3.0 depends on torch>=1.3.1
    torchmetrics 0.2.0 depends on torch>=1.3.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
```

### Expected behavior

`pip` instals all dependencies itself and then installs `torchmetrics`.

### Environment

- PyTorch Version (e.g., 1.0): No
- OS (e.g., Linux): Ubuntu 21.10
- How you installed PyTorch (`conda`, `pip`, source): pip
- Build command you used (if compiling from source): pip install torchmetrics
- Python version: 3.10.2

### PR DESCRIPTION
**Pull request by justusschock**
## What does this PR do?

Fixes #808 
- [x] remove tag for Python 3.10
- [ ] write docs how to use/build on Python 3.10

## Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — CHANGES_REQUESTED** [review]
I would be fine with removing the tag (even I do not like it)
but strongly disagree with a setting limitations for a user using it completely

**Comment by Borda** [comment]
some quick notes on the building custom PT
```bash
pip install -q pyyaml
export USE_CUDA=0  # just to keep it simple
pip install git+https://github.com/pytorch/pytorch.git@v1.10.2
...
```

**Review by Borda — APPROVED** [review]

File: docs/source/pages/quickstart.rst [hunk 1]
```diff
@@ -28,6 +28,19 @@ You can install TorchMetrics using pip or conda:
     # Conda
     conda install -c conda-forge torchmetrics
 
+Eventually if there is missing PyTorch wheel for your OS or Python version you can simply compile `PyTorch from source`_:
+
+.. code-block:: bash
+
+    # Optional ig you do nt need compile GPU support
```
Comment (SkafteNicki) [inline docs/source/pages/quickstart.rst:35]:
```suggestion
    # Optional if you do not need compile GPU support
```

File: docs/source/pages/quickstart.rst [hunk 2]
```diff
@@ -28,6 +28,19 @@ You can install TorchMetrics using pip or conda:
     # Conda
     conda install -c conda-forge torchmetrics
 
+Eventually if there is missing PyTorch wheel for your OS or Python version you can simply compile `PyTorch from source`_:
```
Comment (SkafteNicki) [inline docs/source/pages/quickstart.rst:31]:
```suggestion
Eventually if there is a missing PyTorch wheel for your OS or Python version you can simply compile `PyTorch from source`_:
```

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**